plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1126,7 +1127,13 @@ public void dropPartition(String databaseName, String tableName, List<String> pa
                     .stopOnIllegalExceptions()
                     .run("dropPartition", stats.getDropPartition().wrap(() -> {
                         try (ThriftMetastoreClient client = createMetastoreClient()) {
+                            Partition partition = client.getPartition(databaseName, tableName, parts);
                             client.dropPartition(databaseName, tableName, parts, deleteData);
+                            String partitionLocation = partition.getSd().getLocation();
+                            // TODO https://github.com/trinodb/trino/issues/13821 Add test for 'hive.metastore.thrift.delete-files-on-drop' config property
+                            if (deleteFilesOnDrop && deleteData && !isNullOrEmpty(partitionLocation) && isManagedTable(client.getTable(databaseName, tableName))) {
+                                deleteDirRecursive(hdfsContext, hdfsEnvironment, new Path(partitionLocation));
```

Comment:
I tested this line manually because the I couldn't reproduce the issue in the existing test environment.

Reply:
What about trying to manually reproduce a similar situation as in the issue: https://github.com/trinodb/trino/issues/13017 ?

Reply:
Never mind, I could reproduce the issue in real S3.